### PR TITLE
Add property Transform in Uluru Schema to avoid falsely drifted resources 

### PR DESCRIPTION
**Pull request by anshikg**
*Issue #, if available:*

*Description of changes:* Avoid Falsely drifted resources by adding transformation function for every falsely drifted property

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/resources/test-schema.json [hunk 1]
```diff
@@ -30,6 +30,10 @@
             }
         }
     },
+    "propertyTransform": {
+        "/properties/propertyA": "$join([$string(test), propertyA])",
+        "/properties/propertyB": "$count(propertyB) = 0 ? null : propertyB"
```
Comment (benbridts) [inline src/test/resources/test-schema.json:35]: Out of curiosity:
- which language is this?
- Is that information documented somewhere?

Comment (jotompki) [reply src/test/resources/test-schema.json:35]: Agree with this let's try to document this a bit better in what it actually means. All other schema properties have additional documentation about how exactly they function.

Comment (anshikg) [reply src/test/resources/test-schema.json:35]: Out of curiosity:

which language is this?
> This is jsonta: https://jsonata.org/
Is that information documented somewhere?
> I am working on adding more documentation for this soon. Should be able to update this package's README and the CFN public documentation for the same soon.

File: src/main/resources/schema/provider.definition.schema.v1.json [hunk 2]
```diff
@@ -422,6 +422,14 @@
         "resourceLink": {
             "description": "A template-able link to a resource instance. AWS-internal service links must be relative to the AWS console domain. External service links must be absolute, HTTPS URIs.",
             "$ref": "#/definitions/resourceLink"
+        },
+        "propertyTransform": {
```
Comment (srujithpoondla03) [inline src/main/resources/schema/provider.definition.schema.v1.json:426]: Please add clear description on what this property is used for?

**Review by srujithpoondla03 — CHANGES_REQUESTED** [review]

File: src/main/resources/schema/provider.definition.schema.v1.json (hunk 2 above)
Comment (anshikg) [reply src/main/resources/schema/provider.definition.schema.v1.json:426]: This is part of the documentation and REAME which I am planning to send out next

Comment (srujithpoondla03) [reply src/main/resources/schema/provider.definition.schema.v1.json:426]: My comment is to add the description similar to other properties. For example `resourceLink` property has clear description on what it is used for : "A template-able link to a resource instance. AWS-internal service links must be relative to the AWS console domain. External service links must be absolute, HTTPS URIs."

**Review by ninavolkmuth — APPROVED** [review]

**Review by srujithpoondla03 — APPROVED** [review]

**Pull request merged**